application/src/main/java/org/togetherjava/tjbot/commands/tag/TagManageCommand.java

```diff
@@ -0,0 +1,231 @@
+                        TAG_ID, true),
+                new SubcommandData("createtag", "Create a tag")
+                    .addOption(OptionType.STRING, "id", TAG_ID, true)
+                    .addOption(OptionType.STRING, CONTENT, "Tag content", true),
+                new SubcommandData("createidtag", "Create a tag using a message id")
+                    .addOption(OptionType.STRING, TAG_MINUS_ID, TAG_ID, true)
+                    .addOption(OptionType.NUMBER, MESSAGE_MINUS_ID, "Message id", true));
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandEvent event) {
+        Member member = event.getMember();
+
+        if (!member.hasPermission(Permission.MESSAGE_MANAGE)) {
+            event.reply("You need the MESSAGE_MANAGE permission to use this command!")
+                .setEphemeral(true)
+                .queue();
+
+            return;
+        }
+
+        switch (event.getSubcommandName()) {
+            case "edittag" -> editTag(event);
+            case "editidtag" -> editIdTag(event);
+            case "deletetag" -> deleteTag(event, member);
+            case "createtag" -> createTag(event);
+            case "createidtag" -> createIdTag(event);
+            default -> throw new IllegalStateException();
+        }
+    }
+
+    private void editTag(SlashCommandEvent event) {
+        String tagId = event.getOption("id").getAsString();
+        String content = event.getOption(CONTENT).getAsString();
+
+        if (!tagSystem.exists(tagId)) {
+            event.reply(THIS_TAG_DOES_NOT_EXIST).setEphemeral(true).queue();
+
+            return;
+        }
+
+        tagSystem.put(tagId, content);
+
+        event
+            .replyEmbeds(new EmbedBuilder().setColor(Color.GREEN)
+                .setTitle(SUCCESS)
+                .setTimestamp(LocalDateTime.now())
+                .setFooter(event.getUser().getAsTag())
+                .setDescription("Successfully edited tag '" + tagId + "'")
+                .build())
+            .queue();
+    }
+
+    private void editIdTag(SlashCommandEvent event) {
+        String tagId = event.getOption(TAG_MINUS_ID).getAsString();
+        long messageId = event.getOption(MESSAGE_MINUS_ID).getAsLong();
+
+        if (!tagSystem.exists(tagId)) {
+            event.reply(THIS_TAG_DOES_NOT_EXIST).setEphemeral(true).queue();
+
+            return;
+        }
+
+        event.getMessageChannel().retrieveMessageById(messageId).queue(message -> {
+            tagSystem.put(tagId, message.getContentRaw());
+
+            event
+                .replyEmbeds(new EmbedBuilder().setColor(Color.GREEN)
+                    .setTitle(SUCCESS)
+                    .setTimestamp(LocalDateTime.now())
+                    .setFooter(event.getUser().getAsTag())
+                    .setDescription("Successfully edited tag '" + tagId + "'")
+                    .build())
+                .queue();
+        }, failure -> {
+            if (failure instanceof ErrorResponseException ex
```

Comment:
Does this silently swallow other error conditions?

Reply:
minor but `e` is preferred over `ex` (nice usage of pattern matching btw)